Tools/build/generate_token.py

```diff
@@ -199,23 +206,51 @@ def make_c(infile, outfile='Parser/token.c'):
 
 token_inc_template = f"""\
 .. {AUTO_GENERATED_BY_SCRIPT}
-%s
-.. data:: N_TOKENS
 
-.. data:: NT_OFFSET
+.. list-table::
+   :header-rows: 1
+
+   * - Token
+     - Value
+%s
 """
 
-def make_rst(infile, outfile='Doc/library/token-list.inc'):
+def make_rst(infile, outfile='Doc/library/token-list.inc',
+             rstfile='Doc/library/token.rst'):
     tok_names, ERRORTOKEN, string_to_tok = load_tokens(infile)
     tok_to_string = {value: s for s, value in string_to_tok.items()}
 
+    needs_handwritten_doc = set()
+
     names = []
-    for value, name in enumerate(tok_names[:ERRORTOKEN + 1]):
-        names.append('.. data:: %s' % (name,))
+    for value, name in enumerate(tok_names):
         if value in tok_to_string:
-            names.append('')
-            names.append('   Token value for ``"%s"``.' % tok_to_string[value])
-        names.append('')
+            assert name.isupper()
+            names.append(f'   * - .. data:: {name}')
+            names.append(f'     - ``"{tok_to_string[value]}"``')
+        else:
+            needs_handwritten_doc.add(name)
+
+    has_handwritten_doc = set()
+    with open(rstfile) as fileobj:
+        tokendef_re = re.compile(r'.. data:: (\w+)')
+        for line in fileobj:
+            if match := tokendef_re.fullmatch(line.strip()):
+                if match[1].isupper():
+                    has_handwritten_doc.add(match[1])
```

Comment:
I think we can avoid `re`:
```suggestion
        tokendef_re = re.compile(r'.. data:: (\w+)')
        for line in fileobj:
            if not line.startswith('.. data:: '):
                continue
            tok_name = line.removeprefix('.. data:: ').rstrip()
            if tok_name.isidentifier() and tok_name.isupper():
                has_handwritten_doc.add(tok_name)
```

Reply:
Yes, but at the cost of duplicating the prefix. Is that a good tradeoff?

Reply:
Perhaps, though clarity may be lost in removing duplication:
```suggestion
        for line in map(str.rstrip, fileobj):
            name = line.removeprefix('.. data:: ')
            # Token names must be uppercase and made up of alphanumerics or '_'
            if line != name and name.isidentifier() and name.isupper():
                has_handwritten_doc.add(name)
```

Reply:
I mean, why would I want to avoid `re` when I need to combine matching text and extracting a field?

Reply:
I wasn't a fan of the combination of string methods (`isupper()`) and regular expressions. How's this, which does everything in the regex?
```suggestion
        tokendef_re = re.compile(r'.. data:: ([0-9A-Z_]+)\s*')
        for line in fileobj:
            if match := tokendef_re.fullmatch(line):
                has_handwritten_doc.add(match[1])
```

You could even have:

```suggestion
        tokendef_re = re.compile(r'.. data:: ([0-9A-Z_]+)\s*')
        has_handwritten_doc |= {
            match[1] for line in fileobj
            if (match := tokendef_re.fullmatch(line))
        }
```

Reply:
Ah, I get it now, thanks!
I went for the variant with less magic :)